### PR TITLE
Delete the legacy iOS pixel-firing layer (iOS pixel migration, phase 3)

### PR DESCRIPTION
**Pull request by federicocappelli**
Task/Issue URL: https://app.asana.com/1/137249556945/project/1205842942115003/task/1217858720586294

Last phase of the iOS PixelKit migration, following #6550 which is now merged. Base is `main`

### Description
Now that every call site fires through PixelKit, this deletes the old iOS pixel layer: `Pixel`, `DailyPixel`, `UniquePixel`, `PersistentPixel` and friends, their `PixelFiring` protocols, and the legacy test mocks. `Pixel.Event` and `PixelParameters` stay, in a much smaller `Pixel.swift`.

Three things worth knowing:

- **VPN pixels now use PixelKit's retry queue.** The `PersistentPixelFiring` seams become `PixelKit.fire(..., options: .withRetry)`, and since PixelKit drains its queue after any successful fire, the manual `sendQueuedPixels()` calls are gone. One small difference: `PersistentPixel` stamped `originalPixelTimestamp` on the first send too, PixelKit only adds it on a replay.
- **Fixes a mislabelled bookkeeping key.** A few pixels are meant to fire only once ever, per user. PixelKit records that they've fired by storing a date under a named key, and the migration in this stack copies the equivalent records out of the old `UniquePixel` system into the key named `uniqueByName`. The frequency these pixels use, `.legacyInitial`, declared its key as `"legacyInitial"` while the code actually read and wrote `uniqueByName`. Everything worked, the label was just wrong, so it now matches what the code does. No stored data moves, and macOS picks up the same fix since it's shared PixelKit code.
- **The old `queued-pixels.json` (used for the legacy `PersistentPixelFiring`) no longer has a reader.** Anything still sitting in it at upgrade is dropped. One-off and bounded, since those are already-failed sends, and we left the file in place in case we want to drain it later.

### Testing Steps
1. Green CI.
2. The upgrade path, which CI can't cover because tests always start from an empty store: install a pre-migration build, finish onboarding, upgrade to this one, and check the `_unique` onboarding pixels don't fire again.
3. Connect the VPN and check the retry pixels still send.

#### What could go wrong?
- `.legacyInitial` doesn't enforce the `_u`/`_unique` suffix that `UniquePixel` did. Checked all 18 once-ever fire sites (17 in `OnboardingPixelReporter`, one in `Launching.swift`) and they all end in `_unique`, so nothing starts firing that used to be dropped.
- The 16 VPN retry pixels don't declare `originalPixelTimestamp` / `retriedPixel` in the pixel definitions. Same gap `PersistentPixel` had.

Rollback: revert the PR, nothing downstream depends on it.

---
###### Internal references: [DoD](https://app.asana.com/0/1207634633537039) | [Eng Expectations](https://app.asana.com/0/199064865822552) | [Tech Design Template](https://app.asana.com/0/184709971311943)

---

> [!NOTE]
> **Medium Risk**
> Touches telemetry throttling for upgraded users and replaces VPN retry persistence with PixelKit’s queue; legacy `queued-pixels.json` is no longer replayed on upgrade.
> 
> **Overview**
> Completes the iOS PixelKit migration by **deleting** the legacy firing stack (`Pixel`, `DailyPixel`, `UniquePixel`, `PersistentPixel`, Core `PixelFiring` protocols, related tests/mocks, and Xcode references). **`Pixel.swift` is reduced** to the `Pixel.Event` / `PixelParameters` catalogue plus `Pixel.BuildTarget`; all sends go through PixelKit (`PixelEvent+PixelKit`).
> 
> **VPN and app wiring** drop `PersistentPixelFiring`: `NetworkProtectionTunnelController` injects `(any PixelKitFiring)?` and fires controller start attempt/success/failure with **`options: .withRetry`**. `AppDependencyProvider` and `ReportingService` no longer expose or drain a persistent pixel queue. **Onboarding** drops custom `OnboardingPixelFiring` shims and uses PixelKit directly (`frequency: .legacyInitial` for once-ever events).
> 
> **Shared PixelKit fix:** `.legacyInitial` now **shares the `uniqueByName` throttle map** and records last-fire dates under `.legacyInitial` consistently, so migrated `UniquePixel` state is not ignored and once-ever pixels do not re-fire on upgrade; covered by new unit tests.
> 
> **Docs:** `.cursor/rules/pixels.mdc` and related READMEs updated to state PixelKit-only firing, `PixelKitFiring` / `PixelKitMock` for DI, and that `PixelEvent.swift` is closed to new cases.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit f8cd9ba649d37d205243bc14ab7c885c9626976e. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: iOS/DuckDuckGo/OnboardingFlow/Pixels/OnboardingPixelReporter.swift [hunk 1]
```diff
@@ -185,12 +164,11 @@ final class OnboardingPixelReporter {
...
-    private func fire(event: Pixel.Event, unique: Bool, additionalParameters: [String: String] = [:], includedParameters: [Pixel.QueryParameters] = [.appVersion]) {
-        if unique {
-            uniquePixel.fire(pixel: event, withAdditionalParameters: additionalParameters, includedParameters: includedParameters)
-        } else {
-            pixel.fire(pixel: event, withAdditionalParameters: additionalParameters, includedParameters: includedParameters)
-        }
+    private func fire(event: Pixel.Event, unique: Bool, additionalParameters: [String: String] = [:]) {
+        // `.legacyInitial`: fires once ever, like legacy UniquePixel.fire, but without UniquePixel's
+        // `_u`/`_unique` name-suffix requirement (every onboarding "unique" case ends in `Unique`,
+        // whose name ends in `_unique` - `.uniqueByName` would assert-fail and drop these).
```
[the remark this answers is not in the document]
Comment (federicocappelli) [reply iOS/DuckDuckGo/OnboardingFlow/Pixels/OnboardingPixelReporter.swift]: Trimmed all three in c4d2ccad25: kept the durable facts (why `.legacyInitial` rather than `.uniqueByName`, that PixelKit drains its own retry queue, what the `Pixel` namespace still holds) and dropped the "legacy X used to do Y" framing, including the now-dangling `sendQueuedPixels()` reference.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: iOS/DuckDuckGo/AppServices/ReportingService.swift [hunk 2]
```diff
@@ -181,7 +181,6 @@ final class ReportingService {
             await subscriptionDataReporter.saveWidgetAdded()
         }
         reportFailedCompilationsPixelIfNeeded()
-        AppDependencyProvider.shared.persistentPixel.sendQueuedPixels { _ in }
```
[the remark this answers is not in the document]
Comment (federicocappelli) [reply iOS/DuckDuckGo/AppServices/ReportingService.swift:184]: Confirmed, and not fixed in this PR. After the deletion nothing reads `<AppSupport>/queued-pixels.json`, and nothing writes it either, so the exposure is bounded to pixels that had already failed to send on a pre-upgrade build. PixelKit keeps its retry queue in a separate file (`pixelkit-retry-queue.json`) and `PixelRetryQueueStoring` is internal, so seeding it would mean new public PixelKit API, which is a wider change than this cleanup should carry. Leaving the file in place deliberately rather than deleting it, so a follow-up drain is still possible. Flagged for the pixel owner to decide between draining and accepting the one-time gap.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: .cursor/rules/pixels.mdc [hunk 3]
```diff
@@ -17,7 +17,7 @@ Pixels have the following requirements:
 - Do not use values that are overly precise, e.g. if using an integer value in a parameter, bucket it into ranges rather than including the value verbatim
 - Never include PII, URLs, or other forms of user-identifiable information in pixel names or parameters
 
-For new production iOS code, use PixelKit instead of the legacy iOS `Pixel`, `DailyPixel`, `UniquePixel`, `TimedPixel`, or `PersistentPixel` APIs. Danger exempts only the legacy infrastructure files named by its check, plus test and mock files; these exemptions do not apply to other production code.
+PixelKit is the only pixel-firing API on iOS and macOS. The legacy iOS `Pixel`, `DailyPixel`, `UniquePixel`, `TimedPixel`, and `PersistentPixel` classes, and their `PixelFiring`/`PixelFiringAsync`/`DailyPixelFiring` protocols, have been removed; `PixelFiringMock` and the other legacy test mocks went with them - use `PixelKitMock` instead. `Pixel.Event` and `PixelParameters` survive as the catalogue of already-defined iOS pixel names and parameter keys (see below) - they are the only pieces of the old system still in use.
```
Comment (diegoreymendez) [inline .cursor/rules/pixels.mdc]: I don't think we need to document history.  Just document what's true from today onwards.

File: .cursor/rules/pixels.mdc [hunk 4]
```diff
@@ -49,15 +49,17 @@ pixelKit.fire(uniqueEvent, frequency: .uniqueByName)
 
 ## Pixel Definition Patterns
 
-### Legacy iOS Pixels
+### Existing iOS Pixels (`Pixel.Event`)
 
-Existing legacy iOS pixels are defined as cases on `Pixel.Event` in `iOS/Core/PixelEvent.swift`. Each enum case maps to an HTTP pixel name string via a computed `name` property. This section is reference material for that existing system only.
+Every iOS pixel defined before the PixelKit migration is a case on `Pixel.Event` in `iOS/Core/PixelEvent.swift`. Each enum case maps to an HTTP pixel name string via a computed `name` property, and every one of them fires through PixelKit: `PixelKit.fire(Pixel.Event.someCase, ...)`.
```
Comment (diegoreymendez) [inline .cursor/rules/pixels.mdc]: "the migration" might not be clear to the reader outside of this PR.  We might want to explain Pixel.Event are legacy pixels that relied on a system no longer available.

File: SharedPackages/PixelKit/Sources/PixelKit/RetryQueue/README.md [hunk 5]
```diff
@@ -66,5 +66,4 @@ parameters. The case disappears as those queues age out.
 ## Related
 
 - `PixelKit.Options.retryOnFailure` — the caller-facing switch.
-- `iOS/Core/PersistentPixel.swift` — the older, iOS-only system this is a port of. It is deprecated;
-  new pixels should use PixelKit.
+- iOS's older, `PersistentPixel`-based retry system this was ported from and has since replaced.
```
Comment (diegoreymendez) [inline SharedPackages/PixelKit/Sources/PixelKit/RetryQueue/README.md]: IMHO this should go, no need to keep history in docs.

**Review by diegoreymendez — APPROVED** [review]
This is good to go.

Please see the minor comments I added — mostly just docs cleanup suggestions.

File: iOS/DuckDuckGoTests/OnboardingPixelReporterTests.swift [hunk 6]
```diff
@@ -44,13 +47,12 @@ final class OnboardingPixelReporterTests: XCTestCase {
         sharedPixelHandlerMock = MockOnboardingSharedPixelHandling()
         initSharedPixelsStorageMock()
         MockExperimentPixelFiring.reset()
-        sut = OnboardingPixelReporter(pixel: OnboardingPixelFireMock.self, uniquePixel: OnboardingUniquePixelFireMock.self, statisticsStore: statisticsStoreMock, calendar: calendar, dateProvider: { self.now }, userDefaults: userDefaultsMock, sharedPixelHandler: sharedPixelHandlerMock, sharedPixelsStorage: sharedPixelsStorageMock, downloadReasonExperimentMetric: OnboardingDownloadReasonExperimentMetric(experimentPixelFiring: MockExperimentPixelFiring.self))
```
Comment (federicocappelli) [inline iOS/DuckDuckGoTests/OnboardingPixelReporterTests.swift]: redundant OnboardingPixelFireMock removed here and started using PixelKitMock

File: .cursor/rules/pixels.mdc (hunk 3 above)
Comment (federicocappelli) [reply .cursor/rules/pixels.mdc]: I modified this part but not deleted it, we still need to explain why `Pixel.Event` and `PixelEvents.swift` still exist and why

File: .cursor/rules/pixels.mdc (hunk 4 above)
Comment (federicocappelli) [reply .cursor/rules/pixels.mdc]: reworded

File: SharedPackages/PixelKit/Sources/PixelKit/RetryQueue/README.md (hunk 5 above)
Comment (federicocappelli) [reply SharedPackages/PixelKit/Sources/PixelKit/RetryQueue/README.md]: removed
